### PR TITLE
fix: selection, focus, clipboard for considering rowHeaders

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description
**Fixed**
* selection
  * fixed that keyboard selection with row headers.
  * fixed that mouse selection with row headers.
  * fixed that row header selection with row span.
  * fixed that scroll sync in moving selection on mouse.

* clipboard
  * fixed that clipboard copy and paste with row headers(#489 ).

* keyMap
  * fixed that moving to first cell with row headers.
  * fixed that moving to last cell with row headers.
  * fixed that moving to first column with row headers.
  * fixed that moving to last column with row headers.
  * fixed that moving to first row.
  * fixed that moving to last row.
  * fixed that the selection range on pressing the  `ctrl + a`.

* summary
  * fixed that the summary position with row headers.

* API
  * fixed that `getIndexOfColumn()` method(filtering row header columns).

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/dispatch/keyboard.ts [hunk 1]
```diff
@@ -119,33 +119,33 @@ export function changeSelection(store: Store, command: KeyboardEventCommandType)
...
 
-  changeSelectionRange(selection, inputRange, id);
+    changeSelectionRange(selection, inputRange, id);
+  }
 }
 
 export function removeContent(store: Store) {
   const { column, data } = store;
-  const { visibleColumns } = column;
+  const { visibleColumnsWithRowHeader } = column;
```
Comment (seonim-ryu) [inline src/dispatch/keyboard.ts]: `visibleColumnsWithRowHeader` 위에서 바로 구조분해해도 되겠네요ㅎ
```js
const {
  column: { visibleColumnsWithRowHeader }
  data
} = store;
```

Comment (seonim-ryu) [reply src/dispatch/keyboard.ts]: 아.. `column` 값이 사용되는군요.. ㅠㅠ

File: src/dispatch/mouse.ts [hunk 2]
```diff
@@ -216,28 +215,65 @@ function getFocusCellPos(store: Store) {
...
+  const scrolledPosition = getScrolledPosition(viewInfo, dimension, areaWidth.L);
+
+  return findOffsetIndex(totalColumnOffsets, scrolledPosition.x);
+}
+
+function findRowIndexByPosition(store: Store, viewInfo: ViewInfo) {
+  const { dimension, columnCoords, rowCoords } = store;
+  const { areaWidth } = columnCoords;
+  const scrolledPosition = getScrolledPosition(viewInfo, dimension, areaWidth.L);
+  return findOffsetIndex(rowCoords.offsets, scrolledPosition.y);
```
Comment (seonim-ryu) [inline src/dispatch/mouse.ts]: return문 앞에 개행문자 추가하는건 어떨까요ㅎㅎ

File: src/helper/common.ts [hunk 3]
```diff
@@ -91,8 +91,9 @@ export function findIndexes<T>(predicate: (v: T) => boolean, arr: T[]) {
 
 export function findPrevIndex<T>(arr: T[], predicate: (_: T) => boolean): number {
   const index = findIndex(predicate, arr);
+  const positiveIndex = index - 1 < 0 ? 0 : index - 1;
```
Comment (seonim-ryu) [inline src/helper/common.ts]: `index - 1 < 0` 이 조건을 만족하는 index 값은 0밖에 없으므로 이렇게 처리해도 되지 않을까요??
```js
const positiveIndex = index ? index - 1 : 0; 
```

File: src/query/selection.ts [hunk 4]
```diff
@@ -33,17 +33,19 @@ export function getLeafChildColumnNames(complexHeaderColumns: ComplexColumnInfo[
...
-
-  const startIndex = findPropIndex('name', childNames[0], visibleColumns) - rowHeaderCount;
-  const endIndex =
-    findPropIndex('name', childNames[childNames.length - 1], visibleColumns) - rowHeaderCount;
+  const childNames = sortByVisibleColumns(visibleColumnsWithRowHeader, unsortedChildNames);
+
+  const startIndex = findPropIndex('name', childNames[0], visibleColumnsWithRowHeader);
+  const endIndex = findPropIndex(
+    'name',
+    childNames[childNames.length - 1],
```
Comment (seonim-ryu) [inline src/query/selection.ts]: `childNames.pop()` 사용해도 되겠네요ㅎ

File: src/dispatch/keyboard.ts (hunk 1 above)
Comment (js87zz) [reply src/dispatch/keyboard.ts]: `column`값 사용되는 곳 없어서 말씀하신대로 수정했습니다!

File: src/dispatch/mouse.ts (hunk 2 above)
Comment (js87zz) [reply src/dispatch/mouse.ts]: 여긴 개행을 안했군요.추가하겠습니다!

File: src/helper/common.ts (hunk 3 above)
Comment (js87zz) [reply src/helper/common.ts]: index가 -1 인 경우도 존재하니 아래처럼 변경하겠습니다.
`const positiveIndex = index <= 0 ? 0 : index - 1;`

File: src/query/selection.ts (hunk 4 above)
Comment (js87zz) [reply src/query/selection.ts]: `pop()`은 마지막 요소를 제거하니 사용하면 안 될 것 같습니다!

File: src/helper/rowSpan.ts [hunk 5]
```diff
@@ -77,6 +77,27 @@ export function getRowRangeWithRowSpan(
...
+): Range {
+  if (enableRowSpan(data.sortOptions.columnName)) {
+    const [startRowIndex, endRowIndex] = getMaxRowSpanRange(
+      rowRange,
+      colRange,
+      visibleColumnsWithRowHeader,
+      rowIndex,
+      data
+    );
+    return [startRowIndex, endRowIndex];
```
Comment (seonim-ryu) [inline src/helper/rowSpan.ts]: 바로 반환해도 되겠네요ㅎ
```js
return  getMaxRowSpanRange(
      rowRange,
      colRange,
      visibleColumnsWithRowHeader,
      rowIndex,
      data
    );
```

Comment (js87zz) [reply src/helper/rowSpan.ts]: 넵 반영하겠습니다!

File: src/query/selection.ts (hunk 4 above)
Comment (seonim-ryu) [reply src/query/selection.ts]: 엇 그렇네요 ㄷㄷ 아이고오...

**Pull request merged**